gammapy/catalog/lhaaso.py

```diff
@@ -0,0 +1,190 @@
+        * ``which="WCDA"`` - Sky model for WCDA analysis only.
+
+        If only a limit is given for a parameter it is used as value.
+        Entries not repeated for the second analysis are taken from the first one.
+        """
+
+        pars = {
+            "reference": self._get("E0", which),
+            "amplitude": self._get("N0", which),
+            "index": self._get("gamma", which),
+        }
+
+        errs = {
+            "amplitude": self._get("N0_err", which),
+            "index": self._get("gamma_err", which),
+        }
+
+        model = Model.create("PowerLawSpectralModel", "spectral", **pars)
+
+        for name, value in errs.items():
+            model.parameters[name].error = value
+
+        return model
+
+    def spatial_model(self, which):
+        """Spatial model (`~gammapy.modeling.models.SpatialModel`).
+
+        * ``which="KM2A"`` - Sky model for KM2A analysis only.
+        * ``which="WCDA"`` - Sky model for WCDA analysis only.
+
+        If only a limit is given for a parameter it is used as value.
+        Entries not repeated for the second analysis are taken from the first one.
+        """
+
+        lat_0 = self._get("DECJ2000", which)
+        pars = {"lon_0": self._get("RAJ2000", which), "lat_0": lat_0, "frame": "fk5"}
+
+        pos_err = self._get("pos_err", which)
+        scale_r95 = Gauss2DPDF().containment_radius(0.95)
+
+        errs = {
+            "lat_0": pos_err / scale_r95,
+            "lon_0": pos_err / scale_r95 / np.cos(lat_0),
+        }
+
+        r39, is_ul = self._parse("r39", which)
+        if not is_ul:
+            pars["sigma"] = r39
+            model = Model.create("GaussianSpatialModel", "spatial", **pars)
+        else:
+            model = Model.create("PointSpatialModel", "spatial", **pars)
+
+        for name, value in errs.items():
+            model.parameters[name].error = value
+
+        return model
+
+    @staticmethod
+    def _get_components_geom(models):
+        energy_axis = MapAxis.from_energy_bounds(
+            "0.1 TeV", "2000 TeV", nbin=10, per_decade=True, name="energy"
+        )
+        regions = [m.spatial_model.evaluation_region for m in models]
+        geom = RegionGeom.from_regions(
+            regions, binsz_wcs="0.05 deg", axes=[energy_axis]
+        )
+        return geom.to_wcs_geom()
+
+    def sky_model(self, which="both"):
+        """Sky model (`~gammapy.modeling.models.SkyModel`).
+
+        * ``which="both"`` - Use first model or create a composite template if both models are available.
+        * ``which="KM2A"`` - Sky model for KM2A analysis if available.
+        * ``which="WCDA"`` - Sky model for WCDA analysis if available.
+
+        """
+
+        if which == "both":
+            wcda = self.sky_model(which="WCDA")
+            km2a = self.sky_model(which="KM2A")
+            models = [m for m in [wcda, km2a] if m is not None]
+            if len(models) == 2:
```

Comment:
there is no overlap in energy between the WCDA and KM2A ?

Reply:
no